### PR TITLE
[MNG-7454] Include resolver-transport-http in Maven

### PR DESCRIPTION
**Pull request by cstamas**
Addendum to be2b7f890d98af20eb0753650b6605a68a97ac05:
* fix property name to align with existing ones
* fix logging configuration as no more shaded httpClient
* fix constant names

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
See comments on https://github.com/apache/maven/pull/711#pullrequestreview-942726066

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java [hunk 1]
```diff
@@ -261,23 +261,24 @@ else if ( request.isUpdateSnapshots() )
...
+        else if ( MAVEN_RESOLVER_TRANSPORT_WAGON.equals( transport ) )
         {
             // Make sure (whatever extra priority is set) that wagon is selected
-            configProps.put( WAGON_TRANSPORTER_KEY_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
+            configProps.put( WAGON_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
         }
-        else if ( !MAVEN_TRANSPORT_AUTO.equals( transport ) )
+        else if ( !MAVEN_RESOLVER_TRANSPORT_AUTO.equals( transport ) )
         {
             throw new IllegalArgumentException( "Unknown maven.transport=" + transport
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:279]: Either do `Unknown " + KEY + "=" + value` or `Uknown Maven Resolver transport 'value'".
Remember the pattern we used in Resolver for named lock factories.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java (hunk 1 above)
Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java]: fixed

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java [hunk 2]
```diff
@@ -261,23 +261,24 @@ else if ( request.isUpdateSnapshots() )
...
-            configProps.put( WAGON_TRANSPORTER_KEY_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
+            configProps.put( WAGON_TRANSPORTER_PRIORITY_KEY, RESOLVER_MAX_PRIORITY );
         }
-        else if ( !MAVEN_TRANSPORT_AUTO.equals( transport ) )
+        else if ( !MAVEN_RESOLVER_TRANSPORT_AUTO.equals( transport ) )
         {
-            throw new IllegalArgumentException( "Unknown maven.transport=" + transport
-                    + ". Supported ones are: " + MAVEN_TRANSPORT_WAGON + ", "
-                    + MAVEN_TRANSPORT_RESOLVER + " and " + MAVEN_TRANSPORT_AUTO );
+            throw new IllegalArgumentException( "Unknown " + MAVEN_RESOLVER_TRANSPORT_KEY + "=" + transport
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:279]: This differs from this approach: https://github.com/apache/maven-resolver/pull/168/files#diff-a386807ca89d73dd92c1f453de3af43cee6044c2118d8952ee67ed0e8ba103d6R122-R124

Personally, I like the path from Resolver: Not using the key, but text.

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java]: Done

**Review by michael-o — APPROVED** [review]

**Comment by Aylinee-RH-Arletee** [comment]
d826b7c78ccf3edcb9ea365480c2cfff2332c335

**Comment by jira-importer** [comment]
Resolve #9137

**Pull request merged**